ui/app/adapters/kv/data.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import ApplicationAdapter from '../application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+import { kvId } from 'vault/utils/kv-id';
+import { assert } from '@ember/debug';
+
+export default class KvDataAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  _urlForSecret(backend, path, version) {
+    const base = `${this.buildURL()}/${encodePath(backend)}/data/${encodePath(path)}`;
+    return version ? base + `?version=${version}` : base;
+  }
+
+  createRecord(store, type, snapshot) {
+    const { backend, path, version } = snapshot.record;
+    const url = this._urlForSecret(backend, path);
+    return this.ajax(url, 'POST', { data: this.serialize(snapshot) }).then((resp) => {
+      resp.id = kvId(backend, version, path);
+      return resp;
+    });
+  }
+
+  queryRecord(store, type, query) {
+    const { path, backend, version } = query;
+    return this.ajax(this._urlForSecret(backend, path, version), 'GET').then((resp) => {
+      resp.id = kvId(backend, version, path);
+      return resp;
+    });
+  }
+
+  findRecord(store, type, id) {
+    return this.ajax(`${this.buildURL()}/${id}`, 'GET').then((resp) => {
+      resp.id = id;
+      return resp;
+    });
+  }
+
+  /* Five types of delete operations */
+  deleteRecord(store, type, snapshot) {
+    const { backend, path } = snapshot.record;
+    const { deleteType, deleteVersions } = snapshot.adapterOptions;
+
+    if (!backend || !path) {
+      throw new Error('The request to delete or undelete is missing required attributes.');
+    }
+
+    switch (deleteType) {
+      case 'delete-latest-version':
+        return this.ajax(this._urlForSecret(backend, path), 'DELETE');
+      case 'delete-specific-version':
+        return this.ajax(this._urlForSecret(backend, path), 'POST', {
+          data: { versions: deleteVersions },
+        });
+      case 'destroy-specific-version':
+        return this.ajax(`${this.buildURL()}/${encodePath(backend)}/destroy/${encodePath(path)}`, 'PUT', {
+          data: { versions: deleteVersions },
+        });
+      case 'destroy-everything':
+        return this.ajax(`${this.buildURL()}/${encodePath(backend)}/metadata/${encodePath(path)}`, 'DELETE');
```

Comment:
There is no cross over. And originally this was intended because the delete operations all happen on the data view. However, I just got out of a meeting with design, and that might change. We might move this "destroy-everything" to the metadata view. I'll keep this here and discuss with Chelsea when she returns.